src/subscriptions/restHookLambda/restHook.ts

```diff
@@ -19,15 +20,29 @@ const mergeRequestHeaders = (allowListHeader: string[], channelHeader: string[])
     const mergeHeader = (header: string) => {
         const colonIndex = header.indexOf(':') === -1 ? header.length : header.indexOf(':');
         const headerKey = header.substring(0, colonIndex);
-        const headerValue = header.substring(colonIndex + 1);
-        mergedHeader[headerKey] = headerValue;
+        mergedHeader[headerKey] = header.substring(colonIndex + 1);
     };
 
     allowListHeader.forEach((header) => mergeHeader(header));
     channelHeader.forEach((header) => mergeHeader(header));
     return mergedHeader;
 };
 
+/**
+ * Push latency metric to CloudWatch
+ * @param messages
+ */
+const logLatencyMetric = metricScope((metrics) => async (messages: SubscriptionNotification[]): Promise<void> => {
+    const currentTime = new Date().getTime();
+    messages.forEach((message: SubscriptionNotification) => {
+        metrics.putMetric(
+            'SubscriptionEndToEndLatency',
+            (currentTime - new Date(message.matchedResource.lastUpdated).getTime()) / 1000,
+            Unit.Seconds,
```

Comment:
It's almost always better to emit metrics in the smallest unit available. In this case Milliseconds.

When storing milliseconds you can always divide by 1000 on the graphs if you want to see seconds. But storing the metric as seconds destroys the milliseconds information.